Clinical Key.js

```diff
@@ -0,0 +1,190 @@
+		newItem.url = doc.location.href;
+		newItem.complete();
+	}
+	//if book, use ISBN translator
+	//borrowed from amazon translator
+	else if (contentType == 'book'){
+		var isbn = getElementsByAttr(doc, "data-metadata-isbn", "")[0].getAttribute('data-metadata-isbn');
+		isbn = ZU.cleanISBN(isbn);
+		//use search translator to get metadata from isbn
+		var search = Zotero.loadTranslator("search");
+		//set translators and search
+		search.setHandler("translators", function(obj, translators) {
+			search.setTranslator(translators);
+			search.setHandler("itemDone", function(obj, lookupItem) {
+				newItem=lookupItem;
+				//update ISBN
+				newItem.ISBN = ZU.cleanISBN(isbn);
+				//update url, see NB for rationale why url not used
+				newItem.url = doc.location.href;
+			});
+			search.translate();
+		});
+		//no need to override error handler
+		//save item
+		search.setHandler("done", function() {
+			newItem.complete();
+		});
+		search.setSearch({ ISBN: isbn });
+		search.getTranslators();
+	}
+}
+
+function scrapeBookSection(doc, item){
+	//book title
+	var bookTitle = getElementsByAttr(doc, 'data-once-text','XocsCtrl.title')[0].innerHTML;
+	item.bookTitle = bookTitle;
+	//section title
+	var title  = getElementsByAttr(doc, 'ng-bind-html','ContentCtrl.title')[0].innerHTML;
+	item.title = title;
+	//authors
+	var authorsList  = getElementsByAttr(doc, 'ng-bind-html','XocsCtrl.authorsHtml')[0].getElementsByTagName("a");
+	for (var i = 0;i<authorsList.length;i++){
+		var author = authorsList[i].innerHTML;
+		if(author.includes("<")){
+			author = author.split("<")[0];
+		}
+		item.creators.push(Zotero.Utilities.cleanAuthor(author, 'author'));
+	}
+
+	//page metadata
+	var chapterAndPages = doc.getElementsByClassName("source ng-binding")[0].innerHTML.split("</a>,")[1].trim();
+	//includes both chapter and pages
+	if(chapterAndPages.includes(",")){
+		var chapter = chapterAndPages.split(",")[0].trim();
+		//remove word "chapter"
+		chapter = chapter.toLowerCase();
+		if(chapter.includes("chapter")){
+			chapter = chapter.replace("chapter","").trim();
+		}
+		item.notes.push("Chapter: "+chapter);
+		item.pages = chapterAndPages.split(",")[1].trim();
+	}
+	//only includes pages
+	else{
+		item.pages = chapterAndPages.trim();
+	}
+
+
+	//ISBN metadata
+	var isbn = getElementsByAttr(doc, "data-metadata-isbn", "")[0].getAttribute('data-metadata-isbn');
+	item.ISBN = isbn;
+
+	//edition metadata
+	var edition = getElementsByAttr(doc, 'data-once-text','XocsCtrl.edition')[0].innerHTML.split("Edition")[0].trim();
+	item.edition = edition;
+
+	//publisher metadata
+	var datePub = getElementsByAttr(doc, 'data-once-text','XocsCtrl.copyright')[0].innerHTML;
+	var splitDate = datePub.split(',')[0];
+	var date = splitDate.replace(/\D/g, '');
+	var pub;
+	if (datePub.includes("imprint")){
+		var imprint = datePub.split("by")[1].split(", an")[0].trim();
+		var parent = datePub.split("imprint of")[1].split("Inc.")[0].trim();
```

Comment:
Agreed. Moved date, publisher, and chapter parsing to regex in b7e690b (didn't log multiple commits for clinicalkey)